src/util/pluginUtil.js

```diff
@@ -0,0 +1,54 @@
+const cryptoJS = require('crypto-js');
+const fs = require('fs');
+const path = require('path');
+const fsUtil = require('./fsUtil');
+const logger = require('./logger');
+
+const {
+  ERR_READING,
+} = require('../constants');
+
+const pluginUtil = {
+  /**
+   * Returns the file name for the plugin tag.
+   * Return based on given name if provided, or it will be based on src.
+   * If both are not provided, a md5 generated hash is provided for consistency.
+   */
+  getFileNameForPlugin: (tagAttribs, content) => {
+    if (tagAttribs.name !== undefined) {
+      return `${fsUtil.removeExtension(tagAttribs.name)}.png`;
+    }
+
+    if (tagAttribs.src !== undefined) {
+      const fileName = fsUtil.removeExtension(tagAttribs.src);
+      return `${fileName}.png`;
+    }
+
+    const hashedContent = cryptoJS.MD5(content).toString();
+    return `${hashedContent}.png`;
+  },
+
+  /**
+   * Returns the string content of the plugin.
+   */
+  getPluginContent: ($, element, cwf, config) => {
```

Comment:
Could we expand the scope of this PR a little?
- I realised `cwf` will always be available after refactoring `componentPreprocessor`, let's remove `config` from the entire plantuml plugin, and consequently the entire `config` ( I don't see any other plugins using this, but you could double confirm ). This should be a good step for https://github.com/MarkBind/markbind/pull/714#discussion_r258503158 ( which is no longer necessary with the prior refactors ).

- We currently don't have any plugins that use `config` save for this anymore, so let's repurpose it for plugin utils, as calling `const pluginUtil = require('../../util/pluginUtil');` on every plugin is dependent on the plugin's location. E.g. if the plugin is in `markbind`, one would have to know `pluginUtil`'s location relative to the parent module or in the file system.
  - Let's then expose these pluginUtils. Since we only have one plugin using this now, we can leave it undocumented for a while until we have more and the implementation / requirements becomes more final.

I think its a good step to rectify the solution that was used in https://github.com/MarkBind/markbind/pull/714#discussion_r258503158, and since `pluginUtils` as I see it here isn't parser/page dependent, I think exposing it eventually after a few more rounds of plugin development would be a good option. Might be good to have @marvinchin 's opinion since he was involved though 😁.

Reply:
Hey, could you elaborate more on point 2? Not too sure I am following well

Reply:
I'm thinking of
`preRender: (content, pluginContext, frontMatter, useful safe interfaces (pluginUtil) that we want to expose to plugins) => ...` 
instead of depending on
`preRender: (content, pluginContext, frontMatter, secret unsafe details) => ...` 
or having to rely on `require('../../util/pluginUtil')` where the path may differ from plugin to plugin

Reply:
I see. However, this introduces a breaking change if we remove config from pre/post-render, especially if there are users who have defined their own plugins.

Reply:
It was intentionally undocumented because it exposed too much implementation details https://github.com/MarkBind/markbind/pull/714#discussion_r258418526 ( the above comment ); should be fine here